src/components/AudioAnalyzer/index.js

```diff
@@ -25,10 +28,14 @@ class AudioAnalyzer extends React.Component {
     uiEventEmitter.removeListener('slow-tick', this.draw)
   }
 
+  handleMouseDown (e) {
+    this.isOpen = !this.isOpen
```

Comment:
While it definitely works having `this.isOpen`, all other parts of the app work by taking from the `ui` part of the state. By keeping all state at the top, it means that this little popup can be controlled by other parts of the app. For instance, we could fire an action when the body is clicked to hide it.

In order to do this, I would:

- Create a "container" for AudioAnalyser. This is a redux thing and is a "higher order component" that makes use of `mapStateToProps` and `mapDispatchToProps`. By doing this, you can pass `isOpen` and `onClick` to the component as props.
- To actually make changes to the state, have a look in `store/ui`. Currently there is an `isEditing` property. Up until now it has been used to determine whether popups for editing node titles are showing or hiding. However there's no reason not to use it for this. Something like:

```javascript
onClick: dispatch(uiEditingOpen('audioAnalysisPanel')
```

Then in your container you'd be able to do something like:

```javascript
isOpen: state.ui.isEditing && state.ui.isEditing.type === 'audioAnalysisPanel'
```

If you want to be super neat, write a selector so that it just looks like:
```javascript
isOpen: getUiIsEditingType(state) === 'audioAnalysisPanel'
```

Reply:
You'd probably actually want to make a new action called `uiIsEditingToggle()` and make a reducer test case to handle that.

Reply:
Got the other two comments taken care of, but would you be able to get this last one? Poked around a bit and it's an amount over my head at the moment. Don't know when I'll have the time to trial and error my way into creating a container. I'll definetly be reading your commit though to get a grasp on how it's done in the future.